### PR TITLE
Remove `repo_name` in favor of `repo_id` field

### PR DESCRIPTION
**Pull request by nmiodice**
## All Submissions:
-------------------------------------
* [YES] Have you added an explanation of what your changes do and why you'd like us to include them?
* [YES] I have updated the documentation accordingly.
* [YES] I have added tests to cover my changes.
* [YES] All new and existing tests passed.
* [YES] My code follows the code style of this project.
* [YES] I ran lint checks locally prior to submission.
* [YES] Have you checked to ensure there aren't other open PRs for the same update/change?

## What is the current behavior?
-------------------------------------
There are two fields - `repo_name` and `repo_id` - provided when configuring a build definition to identify the source repository. This change removes the redundant field.

Issue Number: https://github.com/microsoft/terraform-provider-azuredevops/issues/314

## What is the new behavior?
-------------------------------------
Only one field is needed

## Does this introduce a breaking change?
-------------------------------------
- [YES]

The schema for build definitions has changed

### CODE REVIEW

File: azuredevops/resource_build_definition.go [hunk 1]
```diff
@@ -735,14 +730,16 @@ func expandBuildDefinition(d *schema.ResourceData) (*build.BuildDefinition, stri
 	repository := repositories[0].(map[string]interface{})
 
 	repoID := repository["repo_id"].(string)
-	repoName := repository["repo_name"].(string)
+	repoName := ""
 	repoType := RepoType(repository["repo_type"].(string))
```
Comment (EliiseS) [inline azuredevops/resource_build_definition.go:734]: Should this be renamed to repoid then?

File: azuredevops/resource_build_definition.go [hunk 2]
```diff
@@ -735,14 +730,16 @@ func expandBuildDefinition(d *schema.ResourceData) (*build.BuildDefinition, stri
...
 	repoURL := ""
 	if strings.EqualFold(string(repoType), string(RepoTypeValues.GitHub)) {
-		repoURL = fmt.Sprintf("https://github.com/%s.git", repoName)
+		repoURL = fmt.Sprintf("https://github.com/%s.git", repoID)
+		repoName = repoID
 	}
```
Comment (EliiseS) [inline azuredevops/resource_build_definition.go:739]: Why is this being set here in the if instead up where it's set to an empty string?

File: azuredevops/resource_build_definition.go (hunk 1 above)
Comment (nmiodice) [reply azuredevops/resource_build_definition.go:734]: The `repo_id` for GitHub and Bitbucket repositories is the same as the name. For TfsGit repos, however, its an integer ID :/

File: azuredevops/resource_build_definition.go (hunk 2 above)
Comment (nmiodice) [reply azuredevops/resource_build_definition.go:739]: Because its only set for `GitHub` and `Bitbucket` repo types, not `TfsGit`

**Pull request merged**